### DISCUSSION TITLE
Since last update (#82) I can't import to repos that I am a member of only repos that I am an owner of. Was there any change in the last update that could have modified access based on permission level?

### DISCUSSION THREAD

**Discussion by joachimweyl** [gavinr/github-csv-tools#83] [Q&A] [answered]
Both repos are private.
One I am the owner the other I am a member
I was able to import 1.csv to both prior to the last update
I can import 1.csv and even my 900 issue csv to the one I am an owner of.
I cannot import anything to the one I am a member of.
here is the error I get.
```
Error
RequestError [HttpError]: Not Found
    at C:\Users\joach\AppData\Roaming\npm\node_modules\github-csv-tools\node_modules\@octokit\request\dist-node\index.js:86:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Job.doExecute (C:\Users\joach\AppData\Roaming\npm\node_modules\github-csv-tools\node_modules\bottleneck\light.js:405:18) {
  status: 404,
```
I have tested in 3 other repos that I am an owner and I can import to all of them. It appears that if you are nearly a member the import does not work.

Is it possible that the [3rd line](https://github.com/gavinr/github-csv-tools/blob/9919fc67755280b257cbda762ab2f035262d5c05/helpers.js#L3) of `helpers.js` introduced this issue because now it checks with the org and repo and that check requires admin access?

**Comment by joachimweyl** [comment, marked as answer]
I was granted Owner to the second repo and now I can import. it seems like the recent update has removed the ability for GitHub members of a repo to import.

**Reply by gavinr** [reply]
Interesting, so it seems like there's a trade-off here:

1. Use `octokit.issues.create()`  (previous behavior)
    - allows you to create issues in repositories without being repo admin
    - Low rate limiting requirements (~20 or 40 per hour)
2. Use `octokit.request("POST /repos/" + organization + "/" + repository + "/import/issues", ...)` (new way #82)
    - you must be the repo admin
    - Less restrictive rate limits

Would the best behavior be for the script to check if the current user is the repo admin, and then act accordingly?

**Reply by joachimweyl** [reply]
Personally, since I am able to get repo admin I prefer the less restrictive rate. I don't want to speak for others.
Yea, checking if they are a repo admin sounds like a way around it. As long as the documentation warns ahead of time of the restriction for non-admins.

**Reply by joachimweyl** [reply]
just a heads up, I added an issue for this once I realized it was not just me. [issue 84](https://github.com/gavinr/github-csv-tools/issues/84) is for this.
Thank you again for all the work you have done on this tool.
